tests/BuildScriptGeneratorCli.Tests/DetectCommandTest.cs

```diff
@@ -0,0 +1,255 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using Xunit;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Oryx.BuildScriptGenerator.Common;
+using Microsoft.Oryx.BuildScriptGenerator.Node;
+using Microsoft.Oryx.BuildScriptGenerator.Php;
+using Microsoft.Oryx.Detector;
+using Microsoft.Oryx.Tests.Common;
+
+namespace Microsoft.Oryx.BuildScriptGeneratorCli.Tests
+{
+    public class DetectComandTest : IClassFixture<TestTempDirTestFixture>
+    {
+        private readonly string _testDirPath;
+        private readonly TestTempDirTestFixture _testDir;
+
+        public DetectComandTest(TestTempDirTestFixture testFixture)
+        {
+            _testDirPath = testFixture.RootDirPath;
+            _testDir = testFixture;
+        }
+
+        [Fact]
+        public void IsValidInput_UsesCurrentDirectory_IfSourceDirectoryNotSupplied()
+        {
+            // Arrange
+            var detectCommand = new DetectCommand
+            {
+                SourceDir = string.Empty
+            };
+            var testConsole = new TestConsole();
+            var serviceProvider = GetServiceProvider(detectCommand);
+
+            // Act
+            var isValidInput = detectCommand.IsValidInput(serviceProvider, testConsole);
+
+            // Assert
+            Assert.True(isValidInput);
+            Assert.Equal(Directory.GetCurrentDirectory(), detectCommand.SourceDir);
+        }
+
+        [Fact]
+        public void IsValidInput_IsFalse_IfSourceDirectorySuppliedDoesNotExists()
+        {
+            // Arrange
+            var testConsole = new TestConsole();
+            var detectCommand = new DetectCommand
+            {
+                SourceDir = _testDir.GenerateRandomChildDirPath()
+            };
+            var serviceProvider = GetServiceProvider(detectCommand);
+
+            // Act
+            var isValidInput = detectCommand.IsValidInput(serviceProvider, testConsole);
+
+            // Assert
+            Assert.False(isValidInput);
+            Assert.Contains(
+                $"Could not find the source directory",
+                testConsole.StdError);
+        }
+
+        [Fact]
+        public void Execute_OutputsNodePlatformAndNullVersion_WhenJsonFileIsEmpty()
```

Comment:
This name does not sound right...here the version is not detected..similarly fix other test names too